### PR TITLE
refactor Get Logged In User Profile

### PR DESCRIPTION
**Pull request by IngageMaddoxT**

### CODE REVIEW

**Comment by IngageMaddoxT** [comment]
@andycanterbury @IngageKrizJ 
Could each of you review this?  Andy, I'm particularly interested in you review my .net mojo.  I've been looking at this way too long.  Jim, I would love some testing feedback as well.

thanks

File: Gateway/MinistryPlatform.Translation.Test/TranslationTests.cs [hunk 1]
```diff
@@ -222,40 +190,20 @@ public void ShouldNotDeleteResponse()
...
+            Assert.Throws<FaultException<ExceptionDetail>>(
+                () => MinistryPlatformService.DeleteRecord(pageId, responseId, null, token));
         }
 
         [Test]
         public void ShouldReturnTodaysEvents()
         {
             var pageId = Convert.ToInt32(ConfigurationManager.AppSettings["TodaysEventLocationRecords"]);
-            string token = AuthenticationService.authenticate(ConfigurationManager.AppSettings["ApiUser"], ConfigurationManager.AppSettings["ApiPass"]);
+            string token = AuthenticationService.authenticate(ConfigurationManager.AppSettings["ApiUser"],
```
Comment (IngageKrizJ) [inline Gateway/MinistryPlatform.Translation.Test/TranslationTests.cs]: Is this intended to be an "integration" test versus a unit test?  Looks like this is hitting the real MP service.

File: Gateway/MinistryPlatform.Translation.Test/TranslationTests.cs [hunk 2]
```diff
@@ -2,7 +2,6 @@
 using System.Collections.Generic;
```
Comment (IngageKrizJ) [inline Gateway/MinistryPlatform.Translation.Test/TranslationTests.cs]: Looks like these are all integration tests.  I wonder if we can/should exclude these from the CI test runner.

**Comment by IngageKrizJ** [comment]
Tony,
I reviewed, focusing mainly on the tests.  They all look good, I see you
also found out how to have the test expect to throw an exception, I will
keep that in mind for future tests I write.

I did leave one comment on the
Gateway/MinistryPlatform.Translation.Test/TranslationTests.cs
test.  Looks like they are all integration tests, actually hitting MP.  I
wonder if we can exclude these from being run by the CI server.

Anyway, changes look good.

- Jim

On Thu, Mar 26, 2015 at 10:32 PM, Tony Maddox <notifications@github.com>
wrote:

> @andycanterbury <https://github.com/andycanterbury> @IngageKrizJ
> <https://github.com/IngageKrizJ>
> Could each of you review this? Andy, I'm particularly interested in you
> review my .net mojo. I've been looking at this way too long. Jim, I would
> love some testing feedback as well.
>
> thanks
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/crdschurch/crds-angular/pull/142#issuecomment-86790562>
> .
>



-- 
[image: Logo: Ingage Partners, Consulting with Purpose]
<http://www.ingagepartners.com/>

Jim Kriz
Senior Consultant
513.321.7700 (w) *|* 513.703.4218 (m)
2753 Observatory Ave *|* Cincinnati, OH 45208
www.IngagePartners.com <http://www.ingagepartners.com/>
[image: B-Corp Button] <http://www.bcorporation.net/ingagepartners> [image:
Linked-in Button] <http://www.linkedin.com/company/ingage-partners-llc> [image:
Facebook Button]
<http://www.facebook.com/pages/Ingage-Partners/337211479674920>

File: Gateway/MinistryPlatform.Translation.Test/Services/ContactServiceTest.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+using MinistryPlatform.Translation.Services.Interfaces;
+using Moq;
+using NUnit.Framework;
+
+namespace MinistryPlatform.Translation.Test.Services
+{
+    [TestFixture]
+    public class ContactServiceTest
+    {
+        private const int MyProfilePageId = 474;
```
Comment (andycanterbury) [inline Gateway/MinistryPlatform.Translation.Test/Services/ContactServiceTest.cs]: There are several of these hard coded page ids. I remember us saying we didn't want to be dependent on reading the config file for these, but I don't remember if we decided hard coding was the way to go?

Comment (IngageKrizJ) [inline Gateway/MinistryPlatform.Translation.Test/Services/ContactServiceTest.cs]: I feel like the best way to go would be to refactor to allow mocking of
AppSettings, but that is probably a bigger chunk of work than you want to
take on right now.  I suppose an interim step could be to mark AppSettings
as virtual. then extend the class in the test, overriding AppSettings to
behave how you want it, but you'd have to do that in every test (and I'm
also not a big fan of overriding methods in the class being tested...)

On Fri, Mar 27, 2015 at 8:32 AM, andycanterbury <notifications@github.com>
wrote:

> In
> Gateway/MinistryPlatform.Translation.Test/Services/ContactServiceTest.cs
> <https://github.com/crdschurch/crds-angular/pull/142#discussion_r27291186>
> :
>
> > @@ -0,0 +1,66 @@
> > +﻿using System;
> > +using System.Collections.Generic;
> > +using MinistryPlatform.Translation.Services;
> > +using MinistryPlatform.Translation.Services.Interfaces;
> > +using Moq;
> > +using NUnit.Framework;
> > +
> > +namespace MinistryPlatform.Translation.Test.Services
> > +{
> > +    [TestFixture]
> > +    public class ContactServiceTest
> > +    {
> > +        private const int MyProfilePageId = 474;
>
> There are several of these hard coded page ids. I remember us saying we
> didn't want to be dependent on reading the config file for these, but I
> don't remember if we decided hard coding was the way to go?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/crdschurch/crds-angular/pull/142/files#r27291186>.
>



-- 
[image: Logo: Ingage Partners, Consulting with Purpose]
<http://www.ingagepartners.com/>

Jim Kriz
Senior Consultant
513.321.7700 (w) *|* 513.703.4218 (m)
2753 Observatory Ave *|* Cincinnati, OH 45208
www.IngagePartners.com <http://www.ingagepartners.com/>
[image: B-Corp Button] <http://www.bcorporation.net/ingagepartners> [image:
Linked-in Button] <http://www.linkedin.com/company/ingage-partners-llc> [image:
Facebook Button]
<http://www.facebook.com/pages/Ingage-Partners/337211479674920>

**Pull request merged**